lib/Controller/DocumentController.php

```diff
@@ -116,6 +116,13 @@ public function index($fileId) {
 				'instanceId' => $this->settings->getSystemValue('instanceid'),
 			];
 
+			// Update the current file to be accessible with system public shared key
+			$owner = $item->getOwner()->getUID();
+			$absPath = '/' . $owner . '/' .  $item->getInternalPath();
+			$accessList = \OC::$server->getEncryptionFilesHelper()->getAccessList($absPath);
+			$accessList['public'] = true;
+			\OC::$server->getEncryptionManager()->getEncryptionModule()->update($absPath, $owner, $accessList);
```

Comment:
Well, I'm not sure that it is that easy. Because this way you work around the default process how to encrypt/decrypt the file.

Think about following scenario:

The document is already shares with multiple people. Now you want to work with someone else on the document using colabora online and execute the code path in this patch. You start editing and while you work on it someone else with access to the file decides to re-share it. This will trigger a re-calculation of the access list and since the file isn't shared as public link it will remove the additional key again and your edit session will break because access was removed.

I think we need something more integrated in the normal workflow. Either create a real public share for the edit session and remove the share afterwards again or, probably the better solution, implement a small share provider as part of the documents app which introduces it's additional (internal only) share type.